pkg/operator/configobservation/controlplanereplicascount/observe_control_plane_replicas_count.go

```diff
@@ -0,0 +1,89 @@
+package controlplanereplicascount
+
+import (
+	"fmt"
+	"github.com/ghodss/yaml"
+
+	"github.com/openshift/cluster-etcd-operator/pkg/operator/configobservation"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+)
+
+var ControlPlaneReplicasCountPath = []string{"controlPlaneReplicasCount"}
+
+// ObserveControlPlaneReplicasCount sets the current control plane replicas count
+func ObserveControlPlaneReplicasCount(genericListers configobserver.Listers, _ events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, errs []error) {
+	defer func() {
+		// Prune the observed config so that it only contains control-plane-replicas-count field.
+		ret = configobserver.Pruned(ret, ControlPlaneReplicasCountPath)
+	}()
+
+	// read the observed value
+	observedControlPlaneReplicasCountInt, err := readDesiredControlPlaneReplicasCount(genericListers.(configobservation.Listers).ConfigMapListerForKubeSystemNamespace)
+	if err != nil {
+		// we got an error, so we are unable to determine current replicas count
+		return existingConfig, append(errs, err)
+	}
+	// since unmarshalling JSON into an interface value always stores JSON number as a float64 convert the observed value
+	observedControlPlaneReplicasCount := float64(observedControlPlaneReplicasCountInt)
```

Comment:
once we have an observed value, why not just write it and return?

Reply:
right, we could simplify.